docs/IPC.md

```diff
@@ -18,20 +24,296 @@ Note that "persistence" here means surviving the loss of a _process_, not a node
+
+Each change in the table is expected to be updated via an update in the channel.
+
+The `DriverSubscriber` is expected to retrieve the current state of the table, and update the calling
+`pubsub.Subscription` of all changes by sending updates on a channel passed during the `DriverSubscriber`
+initialization.
+
+## Driver Implementations
+
+eve-os currently has one primary driver [socketdriver](https://pkg.go.dev/github.com/lf-edge/eve/pkg/pillar@v0.0.0-20220603153046-23f5ce4eb5ee/pubsub/socketdriver).
+In addition, an [emptydriver](https://github.com/lf-edge/eve/blob/8c6d4ddecf5fec004d4e188f9abc03644c2746aa/pkg/pillar/pubsub/emptydriver.go)
+provides a zero-functionality implementation, which is useful for working with services that require a pubsub but will not be exercising it at all.
+
+Additional implementations may exist in testing, e.g. in-memory drivers.
+### `socketdriver`
+
+The primary `Driver` implemented in eve-os is the [socketdriver](https://pkg.go.dev/github.com/lf-edge/eve/pkg/pillar@v0.0.0-20220603153046-23f5ce4eb5ee/pubsub/socketdriver), which uses:
+
+* data storage via files, one directory per table, one file per key-value entry; it does not use in-memory storage.
+* notification via Unix-domain sockets, one socket per table.
+
+#### Data Storage
+
+The socketdriver stores all data for one table in a specific directory. It does not use in-memory storage.
+This file mechanism is used both for persistent and for non-persistent data. For persistent data,
+it stores files in a directory that will be persisted beyond reboot, whereas for non-persistent data, it stores
+files in an ephemeral directory that will be deleted on reboot.
+
+The root directory upon which socketdriver operates is passed to the socketdriver
+instance upon creation. In the case of normal eve-os operation, that defaults to `/`, but it can be changed upon initialization.
+All other directories are subsidiary to the root directory.
+
+The specific subdirectory used inside the root directory is determined based on the options [here](https://github.com/lf-edge/eve/blob/6160d0e96c72a1954db2a8bdfd99c2fec1972341/pkg/pillar/pubsub/socketdriver/driver.go#L88-L99).
+
+The directory to use and the type of file depends on 2 options:
+
+* `persistent`: whether or not this table should persist, and therefore if files should be in a persisted-beyond-reboot directory or not.
+* `publishToDir`: whether or not this table should publish its data to a directory.
+
+It also includes the "name" of the publication, where `<name>` is calculated:
+
+* global table: `global`
+* `agentScope == ""`: `<agentName>/<topic>`
+* otherwise: `<agentName>/<agentScope>/<topic>`
+
+The above combine to create the location:
+
+1. persistent AND publish: `/persist/config/<name>`
+1. persistent AND NOT publish: `/persist/<name>`
+1. NOT persistent AND publish: `/persist/global/<name>`
+1. !persistent AND !publish: `/var/run/<name>`
```

Comment:
And is there really a different `/var/run` and `/run`? Isn't the first one just symlink to the other?
Oh, maybe with different `RootDir` it might not be...

Reply:
I think it is the same, yes. If it doesn't help, I am as confused about it here as well..

Reply:
/var/run and /run are the same. In the code we should just refer to it as /run

Reply:
Yeah, good point. I can update the docs here.